lib/rspec/matchers/built_in/change.rb

```diff
@@ -372,10 +372,15 @@ def evaluate_value_proc
         end
 
         def inspect_message_with_receiver(message, receiver)
-          if receiver.class == Class
+          case receiver
+          when Module
             "#{receiver}.#{message}"
+          when NilClass
+            "nil##{message}"
```

Comment:
Thank you for waiting. https://github.com/rspec/rspec-support/pull/325 is now merged and `Support.class_of` is available 👌